src/plots.jl

```diff
@@ -1,24 +1,59 @@
 """Draw the given register network.
 
 Requires a Makie backend be already imported."""
-function registernetplot end
+function registernetplot(args...; kwargs...)
```

Comment:
This workaround of using `get_extension` is something I set up for use with types and structs, but it is not a very good way to deal with functions, given the multimethod nature of julia.
Methods of functions can be added in extensions -- that would be much more natural than what we are doing here, of using a new function in a new namespace.

For now we can probably keep this implementation, but it would be important to create an issue to track fixing this hack.

Potentially a better way to do this would be to have a hint attached to method errors. Something like what is done in this PR https://github.com/QuantumSavory/QuantumClifford.jl/pull/416/files# but just warning that the method error happened because the necessary package was not imported.

Reply:
actually, it seems it is worse than that. Currently the way you have namespaced things it looks like a method is being overridden instead of a new function being made. That can be easily fixed by just restricting what is being imported in the namespace and keeping your current implementation.

But given that things need to be fixed anyway, I suggest doing the proper fix with method error hints that I described above.

Here is the errors I am currently getting

```
[ Info: Precompiling QuantumSavoryMakie [92a66160-645b-5231-a29a-c456122f462a] (cache misses: wrong dep version loaded (4), incompatible header (2))
WARNING: Method definition registernetplot(Any...) in module QuantumSavory at /home/stefan/Documents/ScratchSpace/quantumjulia/QuantumSavory.jl/src/plots.jl:4 overwritten in module QuantumSavoryMakie at /home/stefan/.julia/packages/MakieCore/EU17Y/src/recipes.jl:194.
ERROR: Method overwriting is not permitted during Module precompilation. Use `__precompile__(false)` to opt-out of precompilation.
```